google/cloud/documentai_toolbox/wrappers/page.py

```diff
@@ -103,80 +113,90 @@ def sample_table_to_csv():
+    _hocr_bounding_box: Optional[str] = dataclasses.field(init=False, default=None)
 
-
-@dataclasses.dataclass
-class Paragraph:
-    """Represents a wrapped documentai.Document.Page.Paragraph.
-
-    Attributes:
-        documentai_paragraph (google.cloud.documentai.Document.Page.Paragraph):
-            Required. The original google.cloud.documentai.Document.Page.Paragraph object.
-        document_text (str):
-            Required. UTF-8 encoded text in reading order from the document.
-        text (str):
-            Required. UTF-8 encoded text.
-    """
-
-    documentai_paragraph: documentai.Document.Page.Paragraph
-    document_text: dataclasses.InitVar[str]
-    text: str = dataclasses.field(init=False)
-
-    def __post_init__(self, document_text) -> None:
-        self.text = _text_from_layout(
-            layout=self.documentai_paragraph.layout, text=document_text
-        )
+    @property
+    def text(self):
+        if self._text is None:
+            self._text = _text_from_layout(
+                layout=self.documentai_object.layout, text=self._page.text
+            )
+        return self._text
+
+    @property
+    def hocr_bounding_box(self):
+        if not self._hocr_bounding_box:
+            self._hocr_bounding_box = _get_hocr_bounding_box(
+                element_with_layout=self.documentai_object,
+                dimension=self._page.documentai_object.dimension,
+            )
+        return self._hocr_bounding_box
 
 
 @dataclasses.dataclass
 class Line:
     """Represents a wrapped documentai.Document.Page.Line.
 
     Attributes:
-        documentai_line (google.cloud.documentai.Document.Page.Line):
+        documentai_object (google.cloud.documentai.Document.Page.Line):
             Required. The original google.cloud.documentai.Document.Page.Line object.
-        document_text (str):
-            Required. UTF-8 encoded text in reading order from the document.
-        text (str):
-            Required. UTF-8 encoded text.
+        _page (Page):
+            Required. The Page object.
     """
 
-    documentai_line: documentai.Document.Page.Line
-    document_text: dataclasses.InitVar[str]
-    text: str = dataclasses.field(init=False)
+    documentai_object: documentai.Document.Page.Line = dataclasses.field(repr=False)
+    _page: "Page" = dataclasses.field(repr=False)
 
-    def __post_init__(self, document_text) -> None:
-        self.text = _text_from_layout(
-            layout=self.documentai_line.layout, text=document_text
+    tokens: List[Token] = dataclasses.field(
+        init=False, repr=False, default_factory=list
+    )
+    _text: Optional[str] = dataclasses.field(init=False, default=None)
+    _hocr_bounding_box: Optional[str] = dataclasses.field(init=False, default=None)
+
+    def __post_init__(self):
+        self.tokens = _get_children_of_element(
+            self.documentai_object, self._page.tokens
         )
 
+    @property
+    def text(self):
```

Comment:
Eventually let's upgrade `ElementWithLayout` to an abstract base class and move these duplicated codes into it.

Reply:
will do